test/neuron/connection_test.exs

```diff
@@ -0,0 +1,32 @@
+defmodule Neuron.ConnectionTest do
+  use ExUnit.Case
+
+  alias Neuron.Connection
+
+  describe "headers function" do
```

Comment:
This describe should be `build_headers/0`, since that's the function you're testing.